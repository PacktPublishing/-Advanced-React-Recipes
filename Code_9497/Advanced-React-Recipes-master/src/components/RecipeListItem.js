/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/interactive-supports-focus */
import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from './Checkbox';

const RecipeListItem = ({ recipe, favorited, onClick, onFavorited }) => (
  <li
    key={recipe.id}
    onClick={() => onClick(recipe.id)}
    className="py2 border-bottom border-bottom-dashed pointer nowrap"
  >
    <Checkbox
      active={favorited}
      onClick={e => {
        e.stopPropagation();
        onFavorited(recipe.id);
      }}
    />
    <span>{recipe.name}</span>
    <span>{recipe.category}</span>
  </li>
);

RecipeListItem.propTypes = {
  recipe: PropTypes.object,
  favorited: PropTypes.bool,
  onClick: PropTypes.func,
  onFavorited: PropTypes.func,
};

RecipeListItem.defaultProps = {
  recipe: {},
  onClick: () => {},
  onFavorited: () => {},
};

export default RecipeListItem;
