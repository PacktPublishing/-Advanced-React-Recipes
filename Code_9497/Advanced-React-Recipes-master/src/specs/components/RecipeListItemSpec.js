import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import RecipeListItem from '../../components/RecipeListItem';

const testRecipe = {
  id: 1,
  name: 'Test recipe',
  category: 'Dessert',
};

describe('<RecipeListItem />', () => {
  let component;
  let tree;

  test('Should not break if no recipe passed', () => {
    expect(() => renderer.create(<RecipeListItem />)).not.toThrow();
  });

  test('Should render passed recipe', () => {
    component = renderer.create(<RecipeListItem recipe={testRecipe} />);
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorited state', () => {
    component = renderer.create(
      <RecipeListItem recipe={testRecipe} favorited />,
    );
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should call onClick when clicked', () => {
    const onClick = jest.fn();

    component = mount(<RecipeListItem recipe={testRecipe} onClick={onClick} />);
    component.simulate('click');

    expect(onClick.mock.calls.length).toBe(1);
    expect(onClick.mock.calls[0][0]).toBe(testRecipe.id);
  });

  test('Should call onFavorited when favorited', () => {
    const onFavorited = jest.fn();

    component = mount(
      <RecipeListItem recipe={testRecipe} onFavorited={onFavorited} />,
    );
    component.find('div[role="button"]').simulate('click');

    expect(onFavorited.mock.calls.length).toBe(1);
    expect(onFavorited.mock.calls[0][0]).toBe(testRecipe.id);
  });

  test('Should not call onClick when favorited', () => {
    const onClick = jest.fn();
    const onFavorited = jest.fn();

    component = mount(
      <RecipeListItem
        recipe={testRecipe}
        onClick={onClick}
        onFavorited={onFavorited}
      />,
    );
    component.find('div[role="button"]').simulate('click');

    expect(onFavorited.mock.calls.length).toBe(1);
    expect(onClick.mock.calls.length).toBe(0);
  });
});
