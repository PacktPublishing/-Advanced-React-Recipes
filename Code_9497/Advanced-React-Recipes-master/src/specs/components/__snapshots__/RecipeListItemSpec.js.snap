// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem /> Should render favorited state 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer nowrap"
  onClick={[Function]}
>
  <div
    className="inline-block animated"
    onClick={[Function]}
    role="button"
  >
    ✅
  </div>
  <span>
    Test recipe
  </span>
  <span>
    Dessert
  </span>
</li>
`;

exports[`<RecipeListItem /> Should render passed recipe 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer nowrap"
  onClick={[Function]}
>
  <div
    className="inline-block animated"
    onClick={[Function]}
    role="button"
  >
    ⬜️
  </div>
  <span>
    Test recipe
  </span>
  <span>
    Dessert
  </span>
</li>
`;
